### PR TITLE
Dump buffers

### PR DESCRIPTION
**Pull request by dc-secureworks**
Buffer dump option added for both Snort and Suricata engines.

### CODE REVIEW

**Comment by dc-secureworks** [comment]
Pushed a change by Adam that replaced non-printable characters with '.' in the HexDump functions.  Output looks good for that PCAP.

File: app/static/engine-configs/suricata/suricata-6.0.0.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 %YAML 1.1
 ---
-
+# You found it!! /opt/dalton/engine-configs/suricata/suricata-6.0.0.yaml
```
Comment (whartond) [inline app/static/engine-configs/suricata/suricata-6.0.0.yaml:3]: did this get left in on accident?

**Comment by whartond** [comment]
Instead of maintaining a bunch of custom yaml files (I see you updated a number of them in this PR), the recommended strategy is to dynamically update the config file, based on the job settings, when the job is created.  See how it is done here -- https://github.com/secureworks/dalton/blob/bc0541e06bbbe5a11fb4ad581516ec77c0d0fab4/app/dalton.py#L1693-L1712.  You can add code just after that block, before the file gets written out.

**Comment by whartond** [comment]
Can you update the README explaining what this option does?

**Comment by whartond** [comment]
Does this work on older versions of Suricata? If not, you'll want to account for that as necessary, especially in the UI where the "Dump buffers (alerts only) " option would not be shown for unsupported versions.   (As an example, see how the "Use Suricata Socket Control Pcap Processing Mode" Suricata option doesn't show up when unsupported versions are selected.)

File: app/static/engine-configs/suricata/suricata-6.0.0.yaml (hunk 1 above)
Comment (dc-secureworks) [reply app/static/engine-configs/suricata/suricata-6.0.0.yaml]: Yeah, took me a bit of hunting to find where those configs were coming from.  Removed.

**Review by dc-secureworks — COMMENTED** [review]
Gone

**Review by Codingprodigy-tech — APPROVED** [review]

**Comment by dc-secureworks** [comment]
> Instead of maintaining a bunch of custom yaml files (I see you updated a number of them in this PR), the recommended strategy is to dynamically update the config file, based on the job settings, when the job is created. See how it is done here --
> 
> https://github.com/secureworks/dalton/blob/bc0541e06bbbe5a11fb4ad581516ec77c0d0fab4/app/dalton.py#L1693-L1712
> 
> . You can add code just after that block, before the file gets written out.

I'll get this to you this week. Also the README update.

**Review by 0x120102181f0a040a01181c — APPROVED** [review]
Reviewed

**Comment by whartond** [comment]
Everything looks good except now the Suricata confg files (yaml) have had lines removed. Can those be left alone?

**Comment by dc-secureworks** [comment]
> Everything looks good except now the Suricata confg files (yaml) have had lines removed. Can those be left alone?

Restored original commented and disabled lua config in yaml files, verified the dynamic config update function works correctly and the existing/new Lua config is correctly merged.

**Pull request merged**